core/server/worker/src/main/java/alluxio/worker/netty/DataServerBlockReadHandler.java

```diff
@@ -61,9 +65,9 @@
     BlockReadRequestInternal(Protocol.ReadRequest request) throws Exception {
       super(request.getId(), request.getOffset(), request.getOffset() + request.getLength(),
           request.getPacketSize());
-      mBlockReader = mWorker
-          .readBlockRemote(request.getSessionId(), request.getId(), request.getLockId());
-      mWorker.accessBlock(request.getSessionId(), mId);
+      mLockId = mWorker.lockBlock(mSessionId, mId);
```

Comment:
Ideally we can consolidate the ufs and alluxio read, and have the worker do the right thing depending on whether or not the block is in local storage. The downside of this is if the block is removed from the Alluxio worker we choose before the read but still available in other Alluxio workers, but I think it is unnecessary to optimize for this edge case at the moment.